File: 0191-number-of-1-bits/0191-number-of-1-bits.cpp
class Solution {
public:
    int hammingWeight(int n) {
        int count = 0;

        while (n > 0) {
            // Check if the last bit is 1
            if (n & 1)
                count++;

            // Remove the last bit
            n >>= 1;
        }

        return count;
    }
};

File: 0191-number-of-1-bits/0191-number-of-1-bits.py
class Solution:
    def hammingWeight(self, n: int) -> int:
        count = 0

        while n > 0:
            # Check whether the last bit is 1
            if (n & 1) == 1:
                count += 1

            # Right shift to process the next bit
            n >>= 1

        return count